website/content/docs/architecture/scale.mdx

```diff
@@ -0,0 +1,283 @@
+
+Consul can be deployed in either a single datacenter or across multiple datacenters by establishing WAN federation or peering connections. In this context, a datacenter refers to a named environment whose hosts can communicate with low networking latency. Typically, users map a Consul datacenter to a cloud provider region such as AWS us-east-1 or Azure East US.
+
+To ensure consistency and high availability, Consul servers share data using the [Raft consensus protocol](/consul/docs/architecture/consensus). When persisting data, Consul uses Boltdb to store Raft logs and a custom file format for state snapshots. For more information, refer to [Consul architecture](/consul/docs/architecture).
+
+## General deployment recommendations
+
+This section provides general configuration and monitoring recommendations for operating Consul at scale.
+
+### Data plane resiliency
+
+To make service-to-service communication resilient against outages and failures, we recommend spreading multiple service instances for a service across fault domains:
+
+- Spread service instances across _multiple infrastructure-level availability zones_.
+- Spread service instances across _multiple runtime platform instances_, such as multiple Kubernetes clusters.
+- Spread service instances across _multiple Consul datacenters_.
+
+In the event that any individual domain experiences a failure, service failover ensures that healthy instances in other domains remain discoverable. Consul automatically provides service failover between instances within a single [admin partition](/consul/docs/enterprise/admin-partitions) or datacenter.
+
+Service failover across Consul datacenters must be configured in the datacenters before you can use it. Use one of the following methods to configure failover across datacenters:
+
+- **If you are using Consul service mesh**: Implement failover using [service-resolver configuration entries](/consul/docs/connect/config-entries/service-resolver#failover).
+- **If you are using Consul service discovery without service mesh**: Implement [geo-redundant failover using prepared queries](consul/tutorials/developer-discovery/automate-geo-failover).
+
+### Control plane resiliency
+
+When a large number services are deployed to a single datacenter, the Consul servers may experience slower network performance. To make the control plane more resilient against slowdowns and outages, limit the size of individual datacenters by spreading deployments across availability zones, runtimes, and datacenters.
+
+#### Datacenter size
+
+To ensure resiliency, we recommend limiting deployments to a maximum of 5,000 Consul client agents per Consul datacenter. There are two reasons for this recommendation:
```

Comment:
should we mention that this doesn't apply to agentless or can that be assumed? @jkirschner-hashicorp

Reply:
this [blog post](https://www.hashicorp.com/blog/consul-scale-test-report-to-observe-gossip-stability) seems like a great resource to link to from this section -- it talks about Consul scale and gossip stability

it maps performance to actual Consul metrics which is helpful

Reply:
The blog post should go in the "agent gossip management" section.

Maybe something like "Refer to the report on gossip stability at scale for more details."

Reply:
I don't want to assume it's common knowledge externally (especially just after `consul-dataplane`'s initial release on k8s) what the distinction is between Consul client agents and `consul-dataplane`.

I do feel like this is a general problem anywhere in our docs we talk about client agents. I have ideas on that which I can follow up with you on separately...